cdflow_commands/ecs_monitor.py

```diff
@@ -73,6 +74,8 @@ def __init__(self, cluster, environment, component, version, boto_session):
         self._boto_session = boto_session
         self._done = False
         self._seen_ecs_service_events = set()
+        self._new_service_deployment = None
+        self._new_service_deployment_grace_period_count = 0
```

Comment:
I was thinking about this on the way back home and I'm keen to keep the grace period wording, more or less.  I came up with alternative implementation, where I dropped counter and limit variables, introduced a buffer called `_new_service_grace_period` which to start is 60 seconds and as we iterate through events, we subtract from the buffer until it's drained (reaches 0 or goes below).

The reason to keep the wording is I believe it actually fit the purpose quite well, take a look at this excerpt from Wikipedia (definition of grace period):

> A grace period is a period immediately after the deadline for an obligation during which a late fee, or other action that would have been taken as a result of failing to meet the deadline, is waived provided that the obligation is satisfied during the grace period. Grace periods can range from a number of minutes to a number of days or longer, and can apply in situations including arrival at a job, paying a bill, or meeting a government or legal requirement

In our case, this applies to deployment which otherwise would hit the deadline condition early but we give it a bit of slack so we're sure it can meet it's conditions.